pyTST/core.py

```diff
@@ -215,35 +243,68 @@ def onclick(event):
 
                 index = min(np.searchsorted(self.step_time_array, event.xdata), len(self.step_time_array) - 1)
                 update_cursor(index)
+
                 pyplot.draw()
+                global indexGlobal 
+                indexGlobal = index
+                return
+
+
+            hline = ax0.axhline(color='k', lw=0.8, ls='--', alpha=0.6)
+            vline = ax0.axvline(color='k', lw=0.8, ls='--', alpha=0.6)
+
+            text = ax0.text(0.05, 0.9, '', transform=ax0.transAxes)
+            index = update_cursor(np.argmin(self.u95_array[::-1]))
+
+    
+            if selectTimeDiscard:
+              cid = fig0.canvas.mpl_connect('button_press_event', onclick)
 
-            hline = ax.axhline(color='k', lw=0.8, ls='--', alpha=0.6)
-            vline = ax.axvline(color='k', lw=0.8, ls='--', alpha=0.6)
 
-            text = ax.text(0.05, 0.9, '', transform=ax.transAxes)
-            update_cursor(np.argmin(self.u95_array[::-1]))
 
-            cid = fig.canvas.mpl_connect('button_press_event', onclick)
 
 
-        pyplot.ylim(top=np.max(self.u95_array)*2,
+
+        ax0.set_ylim(top=np.max(self.u95_array)*2,
                     bottom= np.min(self.u95_array)/2)
-        pyplot.xlim(right=self.step_time_array[-1]*2,
+        ax0.set_xlim(right=self.step_time_array[-1]*2,
                     left=self.step_time_array[0]/2)
-        pyplot.xlabel("t")
-        pyplot.ylabel("95% uncertainty (u95)")
+        ax0.set_xlabel("timestep",fontsize=16)
+        ax0.set_ylabel("95% uncertainty ($U_{95} (\phi)$)",fontsize=16)
 
+        pyplot.show()
 
-        if filename is None:
-            pyplot.show()
-        else:
-            print("Figure exported to {}".format(filename))
-            pyplot.savefig(filename)
 
-        return fig, ax
+            
+
+        # Plot input signal data
+        fig1, ax1 = pyplot.subplots()
+
+        if selectTimeDiscard:
+          idx = (self.step_time_array.size-indexGlobal)*step_size
+        else:
+          idx = index
 
+        ax1.axvline(self.time_array[idx], color='k', lw=0.8, ls='--', alpha=0.6)
 
+        ax1.plot(self.time_array[0:idx], self.signal_array[0:idx], color='C1',linewidth=2,label='discarded time')
+        ax1.plot(self.time_array[idx:], self.signal_array[idx:], color='C0',linewidth=2,label='stationary region')
+        ax1.plot(self.time_array,self.moving_average, color='r',label='moving average')
 
+        ax1.set_xlabel("$t / $s",fontsize=16)
+        ax1.set_ylabel("$\phi$",fontsize=16)
+        ax1.legend(fontsize=16)
+        
+        
+        
+        if filename is None:
+            pyplot.show()
+        else:
+            pyplot.show()
```

Comment:
shouldn't have .show() here. You want a way to export the tst plot without having any user interaction (for scripting etc.). Or show a user defined component independent of filename